label_studio/io_storages/gcs/utils.py

```diff
@@ -0,0 +1,231 @@
+        return_key: bool = False
+    ):
+        """
+        Iterate files on the bucket. Optionally return limited number of files that match provided extensions
+        :param client: GCS Client obj
+        :param bucket_name: bucket name
+        :param prefix: bucket prefix
+        :param regex_filter: RegEx filter
+        :param limit: specify limit for max files
+        :param return_key: return object key string instead of gcs.Blob object
+        :return: Iterator object
+        """
+        total_read = 0
+        blob_iter = client.list_blobs(bucket_name, prefix=prefix)
+        prefix = str(prefix) if prefix else ''
+        regex = re.compile(str(regex_filter)) if regex_filter else None
+        for blob in blob_iter:
+            # skip dir level
+            if blob.name == (prefix.rstrip('/') + '/'):
+                continue
+            # check regex pattern filter
+            if regex and not regex.match(blob.name):
+                logger.debug(blob.name + ' is skipped by regex filter')
+                continue
+            if return_key:
+                yield blob.name
+            else:
+                yield blob
+            total_read += 1
+            if limit and total_read == limit:
+                break
+
+    @classmethod
+    def _get_signing_kwargs(cls, google_application_credentials):
+        try:
+            cache_key = google_application_credentials
+            credentials = GCS._credentials_cache.get(cache_key)
+            if credentials is None or credentials.expired:
+                credentials, _ = google.auth.default(['https://www.googleapis.com/auth/cloud-platform'])
+                auth_req = google.auth.transport.requests.Request()
+                credentials.refresh(auth_req)
+                GCS._credentials_cache[cache_key] = credentials
+            out = {
+                "service_account_email": credentials.service_account_email,
+                "access_token": credentials.token,
+                "credentials": credentials
+            }
+        except DefaultCredentialsError as exc:
+            logger.error(f"Label studio couldn't load default GCS credentials from env. {exc}",
+                         exc_info=True)
+            out = {}
+        return out
+
+    @classmethod
+    def generate_http_url(
+        cls,
+        url: str,
+        google_application_credentials: Union[str, dict] = None,
+        google_project_id: str = None,
+        presign_ttl: int = 1
+    ) -> str:
+        """
+        Gets gs:// like URI string and returns presigned https:// URL
+        :param url: input URI
+        :param google_application_credentials:
+        :param google_project_id:
+        :param presign_ttl: Presign TTL in minutes
+        :return: Presigned URL string
+        """
+        r = urlparse(url, allow_fragments=False)
+        bucket_name = r.netloc
+        blob_name = r.path.lstrip('/')
+        """Generates a v4 signed URL for downloading a blob.
```

Comment:
looks like we ended up with two docstrings here? Is that intentinoal?

Reply:
@pakelley this was already merged in another PR